src/main/java/sirius/kernel/health/Exceptions.java

```diff
@@ -392,9 +401,29 @@ public static void logDeprecatedMethodUse() {
         StackTraceElement deprecatedMethod = stack[2];
         StackTraceElement caller = stack[3];
         DEPRECATION_LOG.WARN("The deprecated method '%s.%s' was called by '%s.%s'",
-                             deprecatedMethod.getClassName(),
-                             deprecatedMethod.getMethodName(),
-                             caller.getClassName(),
-                             caller.getMethodName());
+                deprecatedMethod.getClassName(),
+                deprecatedMethod.getMethodName(),
+                caller.getClassName(),
+                caller.getMethodName());
+    }
+    
+    /**
+     * Retrieves the actual root {@link Throwable} which ended in the given exception.
+     *
+     * @param e the throwable to begin with
+     * @return the root {@link Throwable} of the given one
+     */
+    public static Throwable getRootCause(@Nullable Throwable e) {
+        if (e == null){
+            return null;
+        }
+
+        Throwable cause = e;
+
+        while (cause.getCause() != null) {
```

Comment:
some exceptions may have themselve as cause (I think..)

Maybe check for that and maybe also add a circuit breaker like "give up after 10 iterations"..